### DISCUSSION TITLE
Skeleton Next - Update 15a - Skeleton Progress Update

### DISCUSSION THREAD

**Discussion by endigo9740** [skeletonlabs/skeleton#2885] [Announcements]
Hey everyone, Chris here from Skeleton 👋

There's been a slew of exciting developments over the past couple weeks. Today we're going to cover a few related to Svelte 5, the new Svelte CLI, and of course Skeleton itself. We've also provided additional standalone posts here:

- https://github.com/skeletonlabs/skeleton/discussions/2886
- https://github.com/skeletonlabs/skeleton/discussions/2887

## Svelte 5 Launched

In case you missed it, Svelte 5 launched at the Svelte Summit over the weekend.

- [Watch the Svelte Summit on YouTube](https://www.youtube.com/watch?v=fAPFsRP-mbc)
- [Svelte 5 Announcement by Rich Harris](https://www.youtube.com/live/fAPFsRP-mbc?si=P-N8XuK2q6pNMzA3&t=12387)
- [Preview the updated Svelte/Kit "omni-site"](https://svelte.dev/)

This is obviously very exciting news for both the Svelte and Skeleton communities. Huge shoutouts to Svelte Society for hosting the event, and of course to the Svelte team for this huge milestone release. After building Skeleton v3 on early versions of Svelte 5 this year, I can safely say this represents a huge leap forward for the framework!

We'll talk more about how this impacts Skeleton in the sections below.

## Svelte CLI

- [Watch the new CLI Announcement](https://www.youtube.com/live/fAPFsRP-mbc?si=vKzwRqZdvML_Mubd&t=11700)

Additionally, we also saw the announcement and launch of the new Svelte CLI. This represents a huge quality of life improvement for Skeleton users, given you can now easily install and configure Tailwind with a single option. This removes a major friction point for scaffolding SvelteKit-based projects for Skeleton. Additionally, support for third party adapters means we could potentially have a one-click install for Skeleton in the near future! (we'll keep you posted on this). Also a huge shoutout to long time Skeleton team member @AdrianGonz97, who handled much of the heavy lifting! Great job Adrian!

## Skeleton Progress Update

With those exciting updates out of the way, let's take a moment to review the state of things for Skeleton as a whole.

### For Skeleton v2

Alongside today's other updates, we've also launched our new [Svelte 5 migration guide for Skeleton v2](https://github.com/skeletonlabs/skeleton/discussions/2887). We encourage you to begin migrating your current Skeleton v2 projects to Svelte 5 as soon as possible, as this will dramatically ease the burden for migrating to Skeleton v3 in the future.

### For Skeleton v3

As you may expect, Svelte 5's launch resolves a major prerequisites for Skeleton v3's launch. It does not, however, mean that Skeleton v3 is quite there yet. That said, we now have an approximate roadmap leading up to launch.

> [!WARNING]  
> As with all development roadmaps, the following is subject to change at any point. However, this should be an accurate representation of the order of operations to reach our launch date.

- Today: in addition to today's progress update (this post), we've also provided a detailed walkthrough of our new [Popover and Modal](https://github.com/skeletonlabs/skeleton/discussions/2886) solutions, as well as a new [Skeleton v2 + Svelte 5 migration guide](https://github.com/skeletonlabs/skeleton/discussions/2640) to help you prepare for v3 in the near future.
- This week: We'll begin implementing the upcoming [Toast component](https://github.com/skeletonlabs/skeleton/issues/2884) and updates to the [latest dependencies](https://github.com/skeletonlabs/skeleton/issues/2888), including the final release of Svelte 5. This will help ensure full compatibility with the final changes.
- Then, in the next release (2 weeks from today), Skeleton v3 will graduate to `Beta` status. This means the project is then functionally feature complete. However there's still plenty to do internally. Look for an upcoming announcement [like this one](https://github.com/skeletonlabs/skeleton/discussions/2681) when the Beta goes live.
- Over the following few weeks, we'll continue testing and implementing onboarding instructions, remaining integration guides, Cookbook sections, and more - anything to help bridge parity to Skeleton v2. Once we're satisfied with these updates, we'll move Skeleton into a public `Release Candidate` status.
- We'll continue testing and refine until we feel everything is production ready. At that point we'll aim to launch the `3.0.0 Release`. Note that we still expect this to drop by end of year.

Please also keep in mind we will be working towards providing a detailed migration guide for updating from Skeleton v2 -> v3. This will be a large effort as there's a lot of changes, but will be made available at or near launch. So please bear with us until then.

## Encouraging Contributions 

I know it's very excited to finally hear the word "release", but I want to make it clear there's still plenty of work left to do! As such, we are actively seeking your contributions to help reach the full release. Help can take many forms:

- Install and test Skeleton v3 in in each stage of it's development (the current alpha, upcoming beta, etc)
- Test and and report bugs and issues for features that are not working as expected
- Review our documentation and let us know of any errors or gaps in the information
- Asking questions about v3 - this helps us identify blindspots in the docs
- Contribute directly by helping us finish any and all [remaining tasks](https://github.com/skeletonlabs/skeleton/milestone/5)

Note that any contribution, no matter how small, can have a huge impact! Really, truly, I appreciate each and every one! If you're interested in contributing, please reach out on Discord or in the comments. You can also refer to our [contribution guidelines](https://tailwindcss.com/docs/guides/nextjs).

**Comment by jdvlpr** [comment]
Thanks for the updates!

**Comment by bn-l** [comment]
How likely is the api of the beta likely to change? If I start using it now will there be a lot of refactoring on release?

Also thanks for the updates!

**Reply by endigo9740** [reply]
@bn-l we reserve the right to modify and implement breaking changes right up until the full release. A full release is a stable release for us. But that said, I don't expect anything drastic. The majority of work remaining is administrative and documentation side.
